data_loader.py

```diff
@@ -6,107 +6,111 @@
 
-        if data_path==None:
+        if data_path == None:
             raise Exception('No data path specified.')
 
         if partition is None:
             raise Exception('Unknown partition type %s.' % partition)
         else:
-            self.partition=partition
+            self.partition = partition
 
-        self.env = lmdb.open(os.path.join(data_path,partition+'_lmdb'), max_readers=1, readonly=True, lock=False,
+        self.env = lmdb.open(os.path.join(data_path, partition + '_lmdb'), max_readers=1, readonly=True, lock=False,
                              readahead=False, meminit=False)
 
-        with open(os.path.join(data_path,partition+'_keys.pkl'),'rb') as f:
+        with open(os.path.join(data_path, partition + '_keys.pkl'), 'rb') as f:
             self.ids = pickle.load(f)
 
-        self.square  = square
+        self.square = square
         self.imgPath = img_path
         self.mismtch = 0.8
-	self.maxInst = 20
+        self.maxInst = 20
 
-	if sem_reg is not None:
-	    self.semantic_reg = sem_reg
+        if sem_reg is not None:
+            self.semantic_reg = sem_reg
         else:
-	    self.semantic_reg = False
+            self.semantic_reg = False
 
         self.transform = transform
         self.target_transform = target_transform
         self.loader = loader
 
     def __getitem__(self, index):
-        recipId  = self.ids[index]
-	# we force 80 percent of them to be a mismatch
-        if self.partition=='train':
-	    match = np.random.uniform() > self.mismtch
-        elif self.partition=='val' or self.partition=='test':
-            match = True 
+        recipId = self.ids[index]
+        # we force 80 percent of them to be a mismatch
+        if self.partition == 'train':
+            match = np.random.uniform() > self.mismtch
+        elif self.partition == 'val' or self.partition == 'test':
+            match = True
         else:
             raise 'Partition name not well defined'
 
-	target = match and 1 or -1
+        target = match and 1 or -1
 
         with self.env.begin(write=False) as txn:
             serialized_sample = txn.get(self.ids[index])
         sample = pickle.loads(serialized_sample)
         imgs = sample['imgs']
 
         # image
-	if target==1:
-            if self.partition=='train':
+        if target == 1:
+            if self.partition == 'train':
                 # We do only use the first five images per recipe during training
-                imgIdx = np.random.choice(range(min(5,len(imgs))))
+                imgIdx = np.random.choice(range(min(5, len(imgs))))
             else:
-                imgIdx = 0 
+                imgIdx = 0
 
-            path = self.imgPath + imgs[imgIdx]['id'] 
-	else:
+            loader_path = [imgs[imgIdx]['id'][i] for i in range(4)]
+            loader_path =  os.path.join(*loader_path)
+            path = os.path.join(self.imgPath, self.partition, loader_path, imgs[imgIdx]['id'])
```

Comment:
`os.path.join(self.imgPath, self.partition, *img_id[:4], img_id)` if you're using python3 :)

Reply:
I'll leave it as is. The rest of the project has some python2 style print functions, this would break consistency. I would suggest it to be included in another change bumping other files to python3 as well